app/js/controllers/actions/membership/membership-list-ctrl.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+angular
+.module('app.controllers')
+.controller('MembershipListCtrl', function($scope, $rootScope, $stateParams, $state, Restangular, apiDescriptor, dataTransformer) {
+	var resourceName = $stateParams.resourceName;
+	var resourceId = $stateParams.id;
+	$scope.resourceName = resourceName;
+	apiDescriptor.then(function(apiDescription) {
+		$scope.rdesc = apiDescription.resource(resourceName);
+	});
+
+
+	var selectionMode = $stateParams.selectionMode;
+	if (!selectionMode || (selectionMode !== 'single ' && selectionMode !== 'multiple')) {
+		selectionMode = 'multiple';
+	}
+	var teamsMap = {};
+	$scope.rolesMap = {};
+	$scope.details = {};
+	$scope.selectionMode = selectionMode;
+	Restangular.all('teams')
+		.getList()
+		.then(function(teams) {
+			_.each(teams, function(element) {
+				teamsMap[element.id] = element.attributes.name;
+			});
+		});
+	Restangular.all(resourceName)
```

Comment:
I think we can simplify all of these requests and condense them to as few as possible. Let's sit down at some point and play around with restangular (https://github.com/mgonto/restangular#methods-description) a bit and see if we make as few calls as possible. Now that we know exactly what we need and have a format for it, we can make it faster!